frontend/src/_components/AppModal.jsx

```diff
@@ -0,0 +1,196 @@
+  const [isNameChanged, setIsNameChanged] = useState(false);
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [clearInput, setClearInput] = useState(false);
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    setIsNameChanged(newAppName?.trim() !== selectedAppName);
+  }, [newAppName, selectedAppName]);
+
+  useEffect(() => {
+    setIsSuccess(false);
+  }, [show]);
+
+  useEffect(() => {
+    inputRef.current?.select();
+  }, [show]);
+
+  useEffect(() => {
+    setIsSuccess(false);
+    setClearInput(false);
+    setNewAppName(selectedAppName);
+  }, [selectedAppName]);
+
+  const handleAction = async (e) => {
+    setDeploying(true);
+    if (!errorText) {
+      setIsLoading(true);
+      try {
+        let success = true;
+        //create app from template
+        if (templateDetails) {
+          success = await processApp(e, newAppName, templateDetails);
+          //import app
+        } else if (fileContent) {
+          success = await processApp(fileContent, newAppName);
+          //rename app/clone existing app
+        } else if (selectedAppId) {
+          success = await processApp(newAppName, selectedAppId);
+          //create app from scratch
+        } else {
+          success = await processApp(newAppName);
+        }
+        if (success === false) {
+          setErrorText('App name already exists');
+          setInfoText('');
+        } else {
+          setErrorText('');
+          setInfoText('');
+          closeModal();
+        }
+      } catch (error) {
+        toast.error(e.error, {
+          position: 'top-center',
+        });
+      }
+    }
+    setIsLoading(false);
+  };
+
+  const handleInputChange = (e) => {
+    const newAppName = e.target.value;
+    console.log(newAppName, newAppName.length);
+    if (newAppName.length == 50) {
+      setInfoText('Maximum length has been reached');
+    } else {
+      setInfoText('');
+      const error = validateAppName(newAppName);
+      setErrorText(error?.errorMsg || '');
+
+      if (clearInput) {
+        setNewAppName('');
+        setClearInput(false);
+      } else {
+        setNewAppName(newAppName);
+      }
+    }
+  };
+
+  return (
+    <Modal
+      show={show}
+      closeModal={closeModal}
+      title={title}
+      footerContent={
+        <>
+          <ButtonSolid variant="tertiary" onClick={closeModal} data-cy="cancel-button" className="modal-footer-divider">
+            Cancel
+          </ButtonSolid>
+          <ButtonSolid
+            onClick={(e) => handleAction(e)}
+            data-cy={actionButton}
+            disabled={
+              isLoading ||
+              errorText ||
+              (actionButton === 'Rename app' &&
+                (!isNameChanged || newAppName.trim().length === 0 || newAppName.length > 50)) || // For rename case
+              (actionButton !== 'Rename app' && (newAppName.length > 50 || newAppName.trim().length === 0)) // For other cases
+            }
```

Comment:
can we convert this condition to a variable?
 
```suggestion
              const createBtnDisableState = isLoading ||
              errorText ||
              (actionButton === 'Rename app' &&
                (!isNameChanged || newAppName.trim().length === 0 || newAppName.length > 50)) || // For rename case
              (actionButton !== 'Rename app' && (newAppName.length > 50 || newAppName.trim().length === 0)) // For other cases
```
```suggestion
              disabled={createBtnDisabledState}
```